cfripper/rules/wildcard_principals.py

```diff
@@ -50,17 +49,19 @@ def check_for_wildcards(self, result: Result, logical_id: str, resource: PolicyD
         for statement in resource._statement_as_list():
             if statement.Effect == "Allow" and statement.principals_with(self.FULL_REGEX):
                 for principal in statement.get_principal_list():
-                    # Check if account ID is allowed
                     account_id_match = self.IAM_PATTERN.match(principal)
-                    if account_id_match:
-                        self.validate_account_id(result, logical_id, account_id_match.group(1))
+                    account_id = account_id_match.group(1) if account_id_match else None
+
+                    # Check if account ID is allowed.
+                    if account_id in self._get_whitelist_from_config():
```

Comment:
Any reason to prefer this over .valid_principals?

```suggestion
                    if account_id in self.valid_principals:
```

Reply:
Yes I think this is the preferred method:

`self.valid_principals` can include other account IDs and the account ID of the stack itself, thus producing too many false negatives. The rule now uses `self._get_whitelist_from_config()`. Code documentation has been added around this, as well as a test to make sure using `self.valid_principals` will result in the template being marked as invalid. 😄